### PR TITLE
Change parameter type

### PR DESCRIPTION
**Pull request by amshinde**

### CODE REVIEW

File: src/shim.c [hunk 1]
```diff
@@ -331,17 +331,17 @@ send_connect_command(struct cc_shim *shim)
...
 
-	if ( fd < 0 || ! buf ) {
+	if ( fd < 0 || ! buf) {
 		return false;
 	}
 
-	do {
-		ret = recv(fd, buf, (size_t)(size-offset), 0);
+	while(offset < size) {
+		ret = recv(fd, buf+offset, size-offset, 0);
```
Comment (dlespiau) [inline src/shim.c:344]: That's actually a behaviour change no? I believe this is indeed a needed fix, but that kind of things shouldn't be buried in a commit that should have no functional changes.

File: src/shim.c [hunk 2]
```diff
@@ -331,17 +331,17 @@ send_connect_command(struct cc_shim *shim)
...
  */
-bool read_wire_data(int fd, uint8_t *buf, ssize_t size)
+bool read_wire_data(int fd, uint8_t *buf, size_t size)
 {
 	ssize_t ret;
-	ssize_t offset = 0;
+	size_t offset = 0;
 
-	if ( fd < 0 || ! buf ) {
+	if ( fd < 0 || ! buf) {
```
Comment (dlespiau) [inline src/shim.c]: This is a spurious white space changes. The rule is no such thing sharing the same commit with other changes. The indentation in the shim is a bit all over the place, seems like we have hard time decided what should be used:

```
if(foo)
if (foo)
if ( foo )
```
and now:
```
if ( foo)
```

Life may well be too short for this, but opened https://github.com/clearcontainers/shim/issues/31 anyway. This isn't high priority.

**Comment by dlespiau** [comment]
Just need an answer to the buf+offset change and if indeed a bug fix, split it in a separate commit.

File: src/shim.c (hunk 1 above)
Comment (sboeuf) [reply src/shim.c:344]: Or at least you should details how you refactored the loop in the commit message.

Comment (dlespiau) [reply src/shim.c:344]: There are two things. The refactoring (including the loop refactor) is one commit. Then there's the `buf+offset`, which is not part of the refactoring as far as I can see (but could be wrong). That's a second commit. But I could be wrong of course.

**Comment by amshinde** [comment]
yes, that indeed is a bug fix that I noticed while refactoring a bit, i'll split it up

**Comment by dlespiau** [comment]
The point was that you could add a const to the first parameter of serialize_frame:

```
uint8_t*
serialize_frame(const struct frame *fr, ssize_t *total_size)
...
```

But doesn't matter that much.

**Comment by dlespiau** [comment]
lgtm

[Approved with PullApprove](https://pullapprove.com/clearcontainers/shim/pull-request/28/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/shim) [Approved with PullApprove](https://pullapprove.com/clearcontainers/shim/pull-request/28/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/shim)

**Comment by dlespiau** [comment]
Sorry, lgtm'ed while there's still a change pending.

**Comment by amshinde** [comment]
@dlespiau Changes pushed

File: src/shim.c [hunk 3]
```diff
@@ -391,11 +391,16 @@ read_frame(struct cc_shim *shim)
 		abort();
 	}
 
-	if (! read_wire_data(shim->proxy_sock_fd, buf, (ssize_t)size)) {
```
Comment (dlespiau) [inline src/shim.c:394]: Was this line removal on purpose?

Comment (amshinde) [reply src/shim.c:394]: shuffled between rewriting commits. fixed

**Pull request merged**